packages/story-editor/src/app/highlights/quickActions/useQuickActions.js

```diff
@@ -94,6 +106,144 @@ const useQuickActions = () => {
       toggleTrimMode,
     })
   );
+  const { resetWithFetch, updateVideoIsMuted, optimizeVideo, optimizeGif } =
+    useLocalMedia(
+      ({
+        actions: {
+          resetWithFetch,
+          updateVideoIsMuted,
+          optimizeVideo,
+          optimizeGif,
+        },
+      }) => {
+        return {
+          resetWithFetch,
+          updateVideoIsMuted,
+          optimizeVideo,
+          optimizeGif,
+        };
+      }
+    );
+
+  const { isTranscodingEnabled } = useFFmpeg();
+
+  // Media Upload Props
+  const allowedMimeTypes = useMemo(() => {
```

Comment:
Curious to know, how `useMemo` would help here?

Reply:
Probably not much.

Looks like a lot of code here (including this bit) is just copied from `packages/story-editor/src/components/library/panes/media/local/mediaPane.js`

Reply:
Yeah, happy to remove if we don't want this.

Reply:
Yeah, let's remove that. Unfortunately, the [official document](https://reactjs.org/docs/hooks-reference.html#usememo) only says that "You may rely on `useMemo` as a performance optimization" but it doesn't elaborate when to use it. Someone on [this article](https://canimerge.com/should-you-really-use-usememo-in-react-lets-find-out/) had done some experiments and the results are interesting. 

So what we learn from that experiment is that `useMemo` comes at a tiny price and if there is not enough complexity, we are doing more harm than good by using it.

![image](https://user-images.githubusercontent.com/6297436/141399705-ecfc6f10-c5d2-4fc2-92a3-3f082a6de616.png)

Reply:
In fact, I have previously thought, that if we remove all unnecessary `useMemo` from the whole project, we should gain some performance improvement.

Reply:
Yeah we're definitely overusing `useMemo` in many places.

Reply:
Surveying use of `useMemo` sounds good. It definitely improved performance for reducing child re-renders in a few places though.

Reply:
Would using `React.memo` for the whole component be a better choice for those special places instead?

Reply:
Ah yea, I was referring to use of `React.memo` not `useMemo`.